Update 02-Teachers-How-to-build-Learning-Units.md (#408)
LDSSA/wiki: docs/_docs/Instructors/02-Teachers-How-to-build-Learning-Units.md

Contents:
- [Academy Moments & Learning Materials](#learning-materials)
  - [Specialization 1 SLUs](#specialization-1-slus)
  - [Specializations 2-6 BLUs](#specializations-2---6)
- [Learning materials](#learning-materials)
  - [Learning Notebooks](#1-learning-notebook)
  - [Example Notebooks](#2-examples-notebook)
  - [Exercise Notebooks](#3-exercise-notebook)
  - [Bootcamp Presentation](#4-bootcamp-presentation)
  - [Hackathon](#5-hackathon)
  - [Hackathon Problem Statement and Dataset](#5.1-hackathon-problem-statement-and-dataset)
  - [Hackathon Evaluation Guidelines](#5.2-hackathon-evaluation-guidelines)
  - [Hackathon Baseline Solution](#5.3-hackathon-instructor-baseline-solution)
- [Development process](#development-process)

---

This page should serve as the source of truth for how to develop a learning unit for the 
Academy. If you are not familiar with the main details about the course, start by checking
out the information [here](../../Starters Academy (LDSSA)/01-Starters-Academy-(Course).md)

# Learning Materials

> The Academy id divided into specializations which are divided into learning units. Each specialization ends

Reviewer: ```suggestion
The Academy is divided into specializations, which are divided into learning units. Each specialization ends
```

* 1 [Examples Notebook](#2-examples-notebook)
* 1 [Exercise Notebook](#3-exercise-notebook)

Each specialization ends with a hackathon (#5-hackathon).

# Learning materials

### 1. Learning Notebook
The Learning Notebook explains the materials in high detail. This notebook 
should be focused on intuitions, and provide motivations, visualizations, 
and explain the new concepts in high granularity. It should work with examples 
to showcase why the students are learning the new concepts.

[Example Learning Notebook](https://github.com/LDSSA/bootcamp/blob/master/units/SLU06_Intermediate_Statistics/Learning%20notebook%20-%20SLU6%20(Intermediate%20statistics).ipynb)

### 2. Examples Notebook

This is rather short and is basically like a cheatsheet for common usage patterns. It can have 
extra examples or applications of the content shown in the learning notebooks.

### 3. Exercise Notebook

The exercise notebook is a series of exercises with auto-grading (see **[how to use nbgrader](https://ldssa.github.io/wiki/Instructors/03-Using-nbgrader-for-Exercise-Notebooks/)** 
for details). This is where the students have to apply the new concepts and so consolidate what they learned. 

> The exercises should make the students

Reviewer: ```suggestion
The exercises should make the students:
```

name format `batchX-instructors`. The README in the repository contains detailed 
instructions around the technical aspects, such as branch naming, setup instructions, 
package requirements, etc.

In short, this is the process:
1. Open the unit tracking issue and an associated PR to indicate the unit is being developed
2. Carefully read the scope of the learning unit in the [Curriculum repo](https://github.com/LDSSA/curriculum-development), and think about what the students should be able to do after they have finished  
3. Go through the learning notebooks and check for anything that needs improvements (check also the open issues)
4. Google around and see how other people teach the concepts. 
5. Submit the Learning Notebook and Example Notebook's improvements  to your PR
6. Go through the Exercise Notebook and define what you will change
7. Change the Exercise Notebook. Remember that students will do this very slowly, so if it doesn't seem absurdly easy, it's probably too hard. 
8. Share with QA.

Recommended execution order whe adding hackathons:
1. Open a tracking issue and an associated PR to indicate the hackathon is being and developed

> 1. Define the category of problem that you think could be interesting to explore  
> 1. Search for available datasets online or in your organization that fit your purpose
> 1. Explore the datasets and pick one that is an interesting challenge
> 1. Define the  target variable 
> 1. Split the data into training and test dataset
> 1. Create the google doc with the problem description 
> 1. Link the document in the README, add the dataset and push to the PR
> 1. Create a set of notebooks that compose the instructor baseline. You can have different notebooks for each
> stage of the process (exploration, cleaning, model). Explain why

Reviewer: ```suggestion
1. Define the category of problem that you think could be interesting to explore  
2. Search for available datasets online or in your organization that fit your purpose
3. Explore the datasets and pick one that is an interesting challenge
4. Define the  target variable 
5. Split the data into training and test dataset
6. Create the google doc with the problem description 
7. Link the document in the README, add the dataset and push to the PR
8. Create a set of notebooks that compose the instructor baseline. You can have different notebooks for each
stage of the process (exploration, cleaning, model). Explain why
```

your baseline is valid - it doesn't need to be the best solution, but it should be a reasonable solution 

> that uses the techniques taught in the unit.
> 1. Create a separate PR and share the baseline solution with QA
> 1. Create the evaluation guidelines in a spreadsheet and share with QA in the PR description

Reviewer: ```suggestion
that uses the techniques taught in the unit.
10. Create a separate PR and share the baseline solution with QA
11. Create the evaluation guidelines in a spreadsheet and share with QA in the PR description
```
